docs/EpochBasedMemoryManagement.md

```diff
@@ -94,35 +107,94 @@ typedef struct _ebpf_epoch_allocation_header
 } ebpf_epoch_allocation_header_t;
 ```
 
-Determining the release epoch is necessarily an expensive operation as
-it requires scanning the epoch of every active execution context, with
-execution contexts being protected by spinlocks. To limit the impact,
-the epoch module uses a one-shot timer to schedule a DPC that computes
-the release epoch by determining the minimum of all execution contexts'
-epochs. The timer is then re-armed when an execution context calls
-ebpf_epoch_exit. The result is that if no execution contexts are active,
-the timer will expire and will not be re-armed.
+DPCs are used to pass messages between CPUs. The following set of messages are
+defined:
+```c code
+typedef enum _ebpf_epoch_cpu_message_type
```

Comment:
My view is that long blocks of code shouldn't be copied out of files (example code is ok) since it can too easily get out of sync.